subsys/bluetooth/controller/ll_sw/ll_filter.c

```diff
@@ -559,9 +620,14 @@ void ll_rl_rpa_update(bool timeout)
 				LL_ASSERT(!err);
 			}
 			if (rl[i].lirk) {
-				err = bt_rpa_create(rl[i].local_irk,
-						    &rl[i].local_rpa);
+				bt_addr_t rpa;
+
+				err = bt_rpa_create(rl[i].local_irk, &rpa);
 				LL_ASSERT(!err);
+				/* thread-safe switch */
```

Comment:
thread-safe may be misleading. May be just explain that code below ensures ISR always has context-safe access to a valid LRPA.